src/component/marker/MarkAreaView.ts

```diff
@@ -123,6 +123,22 @@ function markAreaFilter(coordSys: CoordinateSystem, item: MarkAreaMergedItemOpti
         ) {
             return true;
         }
+        //Directly returning true may also do the work,
+        //because markArea will not be shown automatically
+        //when it's not included in coordinate system.
+        //But filtering ahead can avoid keeping rendering markArea
+        //when there are too many of them.
+        return markerHelper.zoneFilter(coordSys, {
```

Comment:
Another suggestion. There are two places creating same object
```ts
{
    coord: fromCoord,
    x: item.x0,
    y: item.y0
 }
```

We can assign it to a temporary variable ahead to avoid redundant codes.

```ts
const item1 = { coord, x, y }
const item2 = { coord, x, y }
```

Reply:
Right, now it looks neater.